material-ui/index.d.ts

```diff
@@ -561,7 +559,7 @@ declare namespace __MaterialUI {
         targetOrigin?: propTypes.origin;
         textFieldStyle?: React.CSSProperties;
     }
-    export class AutoComplete extends React.Component<AutoCompleteProps, {}> {
+    export class AutoComplete extends React.Component<AutoCompleteProps<any>, {}> {
```

Comment:
Yes it should, but TSX doesn't support generics yet.

Having a generic on the props allows you to hack around this and typecast with the generic.
```tsx
const MyAutoComplete = AutoComplete as React.Commponent<AutoCompleteProps<MyDataItem>, {}>;

render(<MyAutoComplete />);
```